src/apps/replication/lib/replication_failure_detector.cpp

```diff
@@ -117,15 +123,9 @@ void replication_failure_detector::end_ping(::dsn::error_code err, const fd::bea
             {
                 ddebug("update meta server leader to [%s]", ack.this_node.to_string());
                 dsn_group_set_leader(_meta_servers.group_handle(), ack.this_node.c_addr());
```

Comment:
"switch_master" here may not  be necessary, as a later "beacon_ack" will be returned with the proper primary_node and lead to switch of master. But I think add a call to "switch master" will make the code easier to understand, as the "leader of _meta_servers" and the "master in the master_record“ keep the same